### PR TITLE
Use pixi for benchmarking.

### PR DESCRIPTION
**Pull request by kklein**
# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [project]
 name = "metalearners"
 channels = ["conda-forge"]
-platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+platforms = ["linux-64", "osx-arm64", "osx-64"]
```
Comment (kklein) [inline pixi.toml:4]: @pavelzw For some reason `causalml` cannot be installed for `"win-64"`. Do you know of a way to disable this very platform only for the `benchmark` environment?

File: pixi.toml [hunk 2]
```diff
@@ -69,6 +69,22 @@ build = "*"
 [feature.build.tasks]
 build = "python -m build"
 
+[feature.benchmark.dependencies]
```
Comment (pavelzw) [inline pixi.toml:72]:
```suggestion
[feature.benchmark]
platforms = ["linux-64", "osx-arm64", "osx-64"]
[feature.benchmark.dependencies]
```

File: pixi.toml (hunk 1 above)
Comment (pavelzw) [reply pixi.toml:4]:
```suggestion
platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
```

see my comment below

Comment (kklein) [reply pixi.toml]: Thanks a bunch :)

**Review by pavelzw — APPROVED** [review]

File: pixi.toml [hunk 3]
```diff
@@ -69,6 +69,24 @@ build = "*"
...
+numpy = "*"
+pandas = "*"
+lightgbm = "*"
+econml = ">=0.15"
+causalml = "*"
+pip = "*"
+tabulate = "*"
+
+[feature.benchmark.tasks]
+postinstall = "pip install --no-build-isolation --no-deps --disable-pip-version-check -e ."
```
Comment (FrancescMartiEscofetQC) [inline pixi.toml:87]: I'm not sure about this but I think we don't need to redefine this task here, we can just use the default one (as we do in the docs environment)

Comment (kklein) [reply pixi.toml:87]: I thought so, too. Yet this hasn't worked for me:

```bash
pixi run -e benchmark postinstall
postinstall: command not found

Available tasks:
	benchmark
```

Comment (FrancescMartiEscofetQC) [reply pixi.toml:87]: Ah, probably because we have the flag `no-default-feature = true` for the benchmark environment

**Review by FrancescMartiEscofetQC — APPROVED** [review]

**Pull request merged**